examples/ex39.cpp

```diff
@@ -0,0 +1,287 @@
+      attr_sets.AddToBdrAttributeSet("Boundary",
+                                     attr_sets.GetBdrAttributeSet
+                                     ("Southern Boundary"));
+      attr_sets.AddToBdrAttributeSet("Boundary",
+                                     attr_sets.GetBdrAttributeSet
+                                     ("Eastern Boundary"));
+      attr_sets.AddToBdrAttributeSet("Boundary",
+                                     attr_sets.GetBdrAttributeSet
+                                     ("Western Boundary"));
+   }
+
+   // 5. Define a finite element space on the mesh. Here we use continuous
+   //    Lagrange finite elements of the specified order.
+   H1_FECollection fec(order, mesh.Dimension());
+   FiniteElementSpace fespace(&mesh, &fec);
+   cout << "Number of finite element unknowns: "
+        << fespace.GetTrueVSize() << endl;
+
+   // 6. Determine the list of true (i.e. conforming) essential boundary dofs.
+   //    In this example, the boundary conditions are defined by marking all
+   //    the boundary regions corresponding to the boundary attributes
+   //    contained in the set named "ess_name" as essential (Dirichlet) and
+   //    converting them to a list of true dofs.
+   Array<int> ess_tdof_list;
+   if (mesh.bdr_attributes.Size())
+   {
+      Array<int> ess_bdr_marker;
+      mesh.BdrAttrToMarker(attr_sets.GetBdrAttributeSet(ess_name),
+                           ess_bdr_marker);
+      fespace.GetEssentialTrueDofs(ess_bdr_marker, ess_tdof_list);
+   }
+
+   // 7. Set up the linear form b(.) which corresponds to the right-hand side of
+   //    the FEM linear system, which in this case is (1_s,phi_i) where phi_i
+   //    are the basis functions in fespace and 1_s is an indicator function
+   //    equal to 1 on the region defined by the named set "source_name" and
+   //    zero elsewhere.
+   Array<int> source_marker;
+   mesh.AttrToMarker(attr_sets.GetAttributeSet(source_name), source_marker);
+
+   LinearForm b(&fespace);
+   ConstantCoefficient one(1.0);
+   b.AddDomainIntegrator(new DomainLFIntegrator(one), source_marker);
+   b.Assemble();
+
+   // 8. Define the solution vector x as a finite element grid function
+   //    corresponding to fespace. Initialize x with initial guess of zero,
+   //    which satisfies the boundary conditions.
+   GridFunction x(&fespace);
+   x = 0.0;
+
+   // 9. Set up the bilinear form a(.,.) on the finite element space
+   //    corresponding to the Laplacian operator -Delta, by adding the
+   //    Diffusion domain integrator.
+   BilinearForm a(&fespace);
+
+   ConstantCoefficient defaultCoef(1.0e-6);
+   ConstantCoefficient baseCoef(1.0);
+   ConstantCoefficient roseCoef(2.0);
+
+   Array<int> base_marker;
+   Array<int> rose_marker;
+   mesh.AttrToMarker(attr_sets.GetAttributeSet("Base"), base_marker);
+   mesh.AttrToMarker(attr_sets.GetAttributeSet("Rose Even"), rose_marker);
+
+   a.AddDomainIntegrator(new DiffusionIntegrator(defaultCoef));
```

Comment:
Comment that `defaultCoef` is the coefficient in the complement of "Base" and "Rose Even", which is "Rose Odd"?

Reply:
Actually the default coef is applied everywhere but it's so small it doesn't significantly alter diffusion in the other regions. Yes, this is sloppy. In any case more explanation is an excellent idea.